src/js/server/forgot.js

```diff
@@ -0,0 +1,195 @@
+
+require("./lib/password");
+require("./lib/hasMailerHandler");
+require("gpii-handlebars");
+
+fluid.registerNamespace("gpii.express.user.api.forgot.post.handler");
+gpii.express.user.api.forgot.post.handler.checkUser = function (that, user) {
+    if (!user || !user.username) {
+        that.sendResponse(404, { ok: false, message: "No matching user found."});
+    }
+    else {
+        // TODO:  Replace this with a writable dataSource
+        that.user = fluid.copy(user);
+        that.user[that.options.resetCodeKey]  = gpii.express.user.password.generateSalt(that.options.resetCodeLength);
+        that.user[that.options.codeIssuedKey] = new Date();
+
+        var writeUrl = fluid.stringTemplate(that.options.urls.write, { id: that.user._id});
+        var writeOptions = {
+            url:    writeUrl,
+            json:   true,
+            method: "PUT",
+            body:   that.user
+        };
+        request(writeOptions, that.handleRequestResponse);
+    }
+};
+
+gpii.express.user.api.forgot.post.handler.handleRequestResponse = function (that, error, response, body) {
+    if (error) {
+        that.sendResponse(500, { ok: false, message: error.message, stack: error.stack });
+    }
+    else if (response && [200, 201].indexOf(response.statusCode) === -1) {
+        that.sendResponse(response.statusCode, { ok: false, message: body });
+    }
+    else {
+        that.sendMessage();
+    }
+};
+
+fluid.defaults("gpii.express.user.api.forgot.post.handler", {
+    gradeNames: ["gpii.express.user.api.hasMailHandler"],
+    messages: {
+        success: "A password reset code and instructions have been sent to your email address.",
+        error:   "A password reset code could not be sent.  Contact an administrator."
+    },
+    templates: {
+        mail: {
+            text:  "email-forgot-text",
+            html:  "email-forgot-html"
+        }
+    },
+    rules: {
+        mailOptions: {
+            to:      "request.body.email",
+            subject: { literalValue: "Reset your password..."}
+        }
+    },
+    members: {
+        user: null
+    },
+    invokers: {
+        handleRequest: {
+            func: "{reader}.get",
+            args: ["{that}.request.body"]
+        },
+        handleRequestResponse: {
+            funcName: "gpii.express.user.api.forgot.post.handler.handleRequestResponse",
+            args:     ["{that}", "{arguments}.0", "{arguments}.1", "{arguments}.2"] // error, response, body
+        }
+
+    },
+    components: {
+        reader: {
+            type: "gpii.express.user.couchdb.read",
+            options: {
+                url:     "{gpii.express.user.api.forgot}.options.urls.read",
+                termMap: { email: "%email"},
+                rules: {
+                    read: {
+                        "" : "rows.0.value"
+                    }
+                },
+                listeners: {
+                    "onRead.checkUser": {
+                        priority: "last",
```

Comment:
```
 ____            _           _             
| __ ) _   _ ___| |_ ___  __| |            
|  _ \| | | / __| __/ _ \/ _` |            
| |_) | |_| \__ \ ||  __/ (_| |  _   _   _ 
|____/ \__,_|___/\__\___|\__,_| (_) (_) (_)
```

I think I had to do it in some unknown context and then held onto it as some kind of slavish cargo cult thing that had to be done for things to work.  I removed all unneeded priorities throughout and all tests still work.